nixos/modules/programs/steam.nix

```diff
@@ -0,0 +1,27 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.steam;
+in {
+  options.programs.steam = {
+    enable = mkEnableOption "steam";
+  };
+
+  config = mkIf cfg.enable {
+    hardware.opengl = { # this fixes the "glXChooseVisual failed" bug, context: https://github.com/NixOS/nixpkgs/issues/47932
+      enable = true;
+      driSupport32Bit = true;
+    };
+
+    # optionally enable 32bit pulseaudio support if pulseaudio is enabled
+    hardware.pulseaudio.support32Bit = config.hardware.pulseaudio.enable;
+
+    hardware.steam-hardware.enable = true;
```

Comment:
Currently this option only adds some udev rules for "Steam hardware such as the Steam Controller, other supported controllers and the HTC Vive".
Therefore it's relatively lightweight, but if somebody wants to play steam they don't necessarily have the hardware and want that configured.

Should we maybe add an option to enable that manually?

Reply:
Well, you already have this option and can enable it manually.   

I think it shouldn't be enabled by this Nixopt though for the reasons you mentioned.  
if a users searches the Nixopts for `steam`, they'll find this harwdare option aswell and can then decide whether they want it or not.

Reply:
My opinion is one of "just works" matters, additionally the changes are indeed lightweight, so I don't see any problem.

Reply:
I think not enabling `steam-hardware` could be fine, but it would make sense to document that enabling `steam` doesn't enable `steam-hardware`.

Reply:
IMO the best way is to add an option for enabling hardware right in this module, so things are cleaner